dimod/core/structured.py

```diff
@@ -144,3 +145,17 @@ def to_networkx_graph(self):
         G.add_nodes_from(self.nodelist)
 
         return G
+
+    def valid_bqm_graph(self, bqm: "BinaryQuadraticModel"):
```

Comment:
Perhaps a more general approach?
```
def has_subgraph(self, linear, quadratic):
```

I'm fine with either, though.

Reply:
The only reason I'm suggesting this is because we don't mention/specialize for BQM here so far.

Reply:
At least the docstring above mentions "A structured sampler can only sample from binary quadratic models with a specific graph."

Reply:
I do like the generality, but one reason not to is that we can improve the performance if we're using a BQM specifically. I don't mind it being specific to BQMs because at the moment `Sampler` is only defined for BQMs.

Reply:
Ideally, I think, we would add `.samples_from(bqm)` to a `StructuredSampler` class (which we don't have :laughing: )